src/main/scala/firrtl/passes/wiring/Wiring.scala

```diff
@@ -10,157 +10,160 @@ import firrtl.Mappers._
+    * common ancestor (LCA). Verbosely, this modifies the circuit in
+    * the following ways:
+    *   - Adds a pin to each sink module
+    *   - Punches ports up from source signals to the LCA of each sink
+    *   - Punches ports down from LCAs to each sink module
+    *   - Wires sources up to LCA, sinks down from LCA, and across each LCA
+    *
+    * @param c The circuit to modify
+    * @param wi A WiringInfo object containing source, sink, component, and pin information
+    * @return The modified circuit
+    *
+    * @throws WiringException if a sink is equidistant to two sources
     */
   def wire(c: Circuit, wi: WiringInfo): Circuit = {
-    // Split out WiringInfo
-    val source = wi.source
-    val sinks = wi.sinks
-    val compName = wi.comp
-    val pin = wi.pin
-
-    // Maps modules to children instances, i.e. (instance, module)
-    val childrenMap = getChildrenMap(c)
-
-    // Check restrictions
-    val nSources = countInstances(childrenMap, wi.top, source)
-    if(nSources != 1)
-      throw new WiringException(s"Cannot have $nSources instance of $source under ${wi.top}")
-    sinks.foreach { m =>
-      val total = countInstances(childrenMap, c.main, m)
-      val nTop = countInstances(childrenMap, c.main, wi.top)
-      val perTop = countInstances(childrenMap, wi.top, m)
-      if(total != nTop * perTop)
-        throw new WiringException(s"Module ${wi.top} does not contain all instances of $m.")
-    }
+    // Split out wiring info
+    val (ComponentName(compName, ModuleName(source,_)), sinks, pin) = (wi.source, wi.sinks, wi.pin)
+    val iGraph = new InstanceGraph(c)
 
     // Create valid port names for wiring that have no name conflicts
     val portNames = c.modules.foldLeft(Map.empty[String, String]) { (map, m) =>
       map + (m.name -> {
         val ns = Namespace(m)
-        if(sinks.contains(m.name)) ns.newName(pin)
+        if(sinks.filter(namedName(_) == m.name).nonEmpty) ns.newName(pin)
         else ns.newName(tokenize(compName) filterNot ("[]." contains _) mkString "_")
       })
     }
 
-    // Create a lineage tree from children map
-    val lineages = getLineage(childrenMap, wi.top)
-
-    // Populate lineage tree with relationship information, i.e. who is source,
-    //   sink, parent of source, etc.
-    val withFields = setSharedParent(wi.top)(setFields(sinks, source)(lineages))
-
-    // Populate lineage tree with what to instantiate, connect to/from, etc.
-    val withThings = setThings(portNames, compName)(withFields)
-
-    // Create a map from module name to lineage information
-    val map = pointToLineage(withThings)
-
     // Obtain the source component type
     val sourceComponentType = getType(c, source, compName)
+    val sinkComponents = sinks
+      .foldLeft(Map.empty[String, String]) { (map, m) =>
+        map ++ { m match {
+          case ComponentName(c, ModuleName(m, _)) => Map(m -> c)
+          case _ => Map()
+        }}}
```

Comment:
I think this could be better written as just
```scala
val sinkComponents = sinks.collect { case ComponentName(c, ModuleName(m, _)) => m -> c }.toMap
```